builder/vmware/common/driver_esx5.go

```diff
@@ -670,17 +670,28 @@ func (d *ESX5Driver) Download(src, dst string) error {
 	return d.comm.Download(d.datastorePath(src), file)
 }
 
-func (d *ESX5Driver) VerifyChecksum(ctype string, hash string, file string) bool {
-	if ctype == "none" {
+func (d *ESX5Driver) VerifyChecksum(hash string, file string) bool {
+	if hash == "none" {
 		if err := d.sh("stat", strconv.Quote(file)); err != nil {
 			return false
 		}
-	} else {
-		stdin := bytes.NewBufferString(fmt.Sprintf("%s  %s", hash, file))
-		_, err := d.run(stdin, fmt.Sprintf("%ssum", ctype), "-c")
-		if err != nil {
-			return false
-		}
+		return true
+	}
+
+	// TODO(azr): this will not work with checksums from files
+	vs := strings.SplitN(hash, ":", 2)
+	var ctype string
+	switch len(vs) {
+	case 2:
+		ctype, hash = vs[0], vs[1]
+	default:
+		return false
+	}
+
+	stdin := bytes.NewBufferString(fmt.Sprintf("%s  %s", hash, file))
+	_, err := d.run(stdin, fmt.Sprintf("%ssum", ctype), "-c")
```

Comment:
Here, `filesum` is not a command and this depends on the existence of hash verification function on the system; would it be better to be able to run the checksumming using the go-getter ? 
Should we just err when we see a `file` type ?

Reply:
Do you have any thoughts on this @SwampDragons ?